modules/monitoring/library/Monitoring/Command/Renderer/IcingaApiCommandRenderer.php

```diff
@@ -156,25 +158,40 @@ public function renderScheduleDowntime(ScheduleServiceDowntimeCommand $command)
             'trigger_name'  => $command->getTriggerId()
         );
         $commandData = $data;
+
         if ($command instanceof PropagateHostDowntimeCommand) {
-            /** @var \Icinga\Module\Monitoring\Command\Object\PropagateHostDowntimeCommand $command */
             $commandData['child_options'] = $command->getTriggered() ? 1 : 2;
+        } elseif ($command instanceof ApiScheduleHostDowntimeCommand) {
+            // We assume that it has previously been verified that the Icinga version is
+            // equal to or greater than 2.11.0
+            $commandData['child_options'] = $command->getChildOptions();
+        }
+
+        $allServicesCompat = false;
+        if ($command instanceof ScheduleHostDowntimeCommand) {
+            if ($command->isForAllServicesNative()) {
+                // We assume that it has previously been verified that the Icinga version is
+                // equal to or greater than 2.11.0
+                $commandData['all_services'] = $command->getForAllServices();
+            } else {
+                $allServicesCompat = $command->getForAllServices();
+            }
         }
+
         $this->applyFilter($commandData, $command->getObject());
         $apiCommand = IcingaApiCommand::create($endpoint, $commandData);
-        if ($command instanceof ScheduleHostDowntimeCommand
-            /** @var \Icinga\Module\Monitoring\Command\Object\ScheduleHostDowntimeCommand $command */
-            && $command->getForAllServices()
-        ) {
-            $commandData = $data + array(
-                'type'          => 'Service',
-                'filter'        => 'host.name == host_name',
-                'filter_vars'   => array(
-                    'host_name' => $command->getObject()->getName()
-                )
-            );
+
+        if ($allServicesCompat) {
+            $commandData = $data + [
+                    'type'        => 'Service',
+                    'filter'      => 'host.name == host_name',
+                    'filter_vars' => [
+                        'host_name' => $command->getObject()->getName()
+                    ]
+                ];
```

Comment:
This looks too far indented, but the checks don't fail :man_shrugging:

Reply:
Yeah I though so too, but PHPStorm also always reformats that section.

Reply:
Don't get me going with PHPStorm and formatting indentation, that's broken for me since some time.

Reply:
Do you want me to change it?

Reply:
I'd do it, but again, the checks don't mind. So it's just my opinion.

Reply:
Done.